README.md
[![npm](https://img.shields.io/npm/dt/react-chat-elements.svg?label=npm-install&style=flat)]() [![Github pre-)release](https://img.shields.io/github/release-pre/Detaysoft/react-chat-elements.svg?style=flat)]()

# react-chat-elements

## Reactjs chat elements

## **[Full Documentation](https://detaysoft.github.io/docs-react-chat-elements/)**, **[Changelog](https://github.com/Detaysoft/react-chat-elements/blob/master/CHANGELOG.md)**

## Install

```
npm install react-chat-elements --save
```

### For React Native (Experimental)

Step 1: Install react-chat-elements

```
npm install react-chat-elements --save
```

If you have already installed react-native-vector-icons as a dependency for your project you can skip this step. Otherwise run the following command:

Step 2: Install react-native-vector-icons

```
npm i --save react-native-vector-icons
```

**Link:**

```
react-native link react-native-vector-icons
```

_If you have any issues installing react-native-vector-icons, check out their installation guide [here](https://github.com/oblador/react-native-vector-icons#installation)._

### Imports

```javascript
// RCE CSS
import 'react-chat-elements/dist/main.css'
// MessageBox component
import { MessageBox } from 'react-chat-elements'
```

#### React Native

```javascript
// MessageBox component
import { MessageBox } from 'react-chat-elements/native'
```

## Components

1. [ChatItem](#chatitem-component)
2. [MessageBox](#messagebox-component)
3. [SystemMessage](#systemmessage-component)
4. [MessageList](#messagelist-component)
5. [ChatList](#chatlist-component)
6. [Input](#input-component)
7. [Button](#button-component)
8. [Popup](#popup-component)
9. [SideBar](#sidebar-component)
10. [Navbar](#navbar-component)
11. [Dropdown](#dropdown-component)
12. [Avatar](#avatar-component)
13. [LocationMessage](#locationmessage-component)
14. [SpotifyMessage](#spotifymessage-component)
15. [MeetingItem](#meetingitem-component)
16. [MeetingList](#meetinglist-component)
17. [MeetingLink](#meetinglink-component)

## ChatItem Component

![chatitem-photo](https://user-images.githubusercontent.com/15075759/29243612-d7f45ce0-7fab-11e7-96ce-7792e861d6aa.png)

```javascript
import { ChatItem } from 'react-chat-elements'
;<ChatItem
  avatar={'https://facebook.github.io/react/img/logo.svg'}
  alt={'Reactjs'}
  title={'Facebook'}
  subtitle={'What are you doing?'}
  date={new Date()}
  unread={0}
/>
```

#### ChatItem props

| prop             | default | type       | description                                            |
| ---------------- | ------- | ---------- | ------------------------------------------------------ |
| avatar           | none    | string     | ChatItem avatar photo url                              |
| avatarFlexible   | false   | boolean    | flexible ChatItem avatar photo                         |
| alt              | none    | string     | ChatItem avatar photo alt text                         |
| title            | none    | string     | ChatItem title                                         |
| subtitle         | none    | string     | ChatItem subtitle                                      |
| date             | none    | date       | ChatItem date                                          |
| dateString       | none    | string     | ChatItem represents dateString or timeagojs(now, date) |
| unread           | 0       | int        | ChatItem unread count                                  |
| onClick          | none    | function   | ChatItem on click                                      |
| onContextMenu    | none    | function   | ChatItem on context menu                               |
| statusColor      | none    | color      | ChatItem status color                                  |
| statusColorType  | badge   | string     | ChatItem status color type (encircle, badge)           |
| statusText       | none    | color      | ChatItem status text                                   |
| lazyLoadingImage | none    | image path | lazy loading image                                     |
| muted            | false   | bool       | chat mute info                                         |
| showMute         | false   | bool       | chat mute button visibilty                             |
| showVideoCall    | false   | bool       | chat video call button visibilty                       |
| onClickMute      | none    | function   | mute button                                            |
| onClickVideoCall | none    | function   | video call button                                      |

## MessageBox Component

| file                                                                                                                  | photo                                                                                                                  | text                                                                                                                  | location                                                                                                                  | video                                                                                                     | audio                                                                                                      |
| --------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- |
| ![file-message](https://user-images.githubusercontent.com/15075759/29243524-786baf8c-7fa9-11e7-92b2-3960cc34fcf4.png) | ![photo-message](https://user-images.githubusercontent.com/15075759/29243556-1368c4e8-7faa-11e7-9fdd-57b67542c381.png) | ![text-message](https://user-images.githubusercontent.com/15075759/29243573-a809456e-7faa-11e7-97b4-d258bc6a7728.png) | ![location-message](https://user-images.githubusercontent.com/15075759/30582943-2b3338aa-9d2e-11e7-93d5-8614563b9217.png) | ![](https://user-images.githubusercontent.com/15075759/98530021-74ca1300-228f-11eb-88ab-5ae5ae33616d.png) | ![](https://user-images.githubusercontent.com/41473129/105960654-98327c00-608e-11eb-9ac4-5b014974b87c.png) |

```javascript
import { MessageBox } from 'react-chat-elements'
;<MessageBox
  position={'left'}
  type={'photo'}
  text={'react.svg'}
  data={{
    uri: 'https://facebook.github.io/react/img/logo.svg',
    status: {
      click: false,
      loading: 0,
    },
  }}
/>
```

#### MessageBox props

| prop                    | default    | type                 | description                                                                                        |
| ----------------------- | ---------- | -------------------- | -------------------------------------------------------------------------------------------------- |
| id                      | i (index)  | string               | message box id                                                                                     |
| position                | left       | string               | message box position                                                                               |
| type                    | text       | string               | message type (text, photo, file, location, spotify, video, audio)                                  |
| text                    | none       | string               | message text                                                                                       |
| title                   | none       | string               | message title                                                                                      |
| titleColor              | none       | string(color)        | message title color                                                                                |
| data                    | {}         | object               | message data                                                                                       |
| date                    | new Date() | Date                 | message date                                                                                       |
| dateString              | none       | string               | message represents dateString or timeagojs(now, date)                                              |
| onClick                 | none       | function             | message on click (message(object) is returned)                                                     |
| onOpen                  | none       | function             | message on open (file or photo) (message(object) is returned)                                      |
| onDownload              | none       | function             | message on download (file or photo) (message(object) is returned)                                  |
| onLoad                  | none       | function             | message on load photo                                                                              |
| onPhotoError            | none       | function             | message on error photo                                                                             |
| onTitleClick            | none       | function             | message title on click event                                                                       |
| onForwardClick          | none       | function             | message forward on click event                                                                     |
| onReplyClick            | none       | function             | message reply on click event                                                                       |
| onMeetingMessageClick   | none       | function             | meeting message on click event                                                                     |
| onMeetingTitleClick     | none       | function             | meeting title message on click event                                                               |
| onMeetingVideoLinkClick | none       | function             | meeting video link message on click event                                                          |
| onReplyMessageClick     | none       | function             | reply message on click event                                                                       |
| onRemoveMessageClick    | none       | function             | message remove on click event                                                                      |
| onMeetingMoreSelect     | none       | function             | message list item onMeetingMoreSelect event, gets 3 parameters: message item, index of item, event |
| onMeetingLinkClick      | none       | function             | meeting link on click event                                                                        |
| onContextMenu           | none       | function             | message contextmenu click event                                                                    |
| forwarded               | none       | boolean              | message forward icon                                                                               |
| replyButton             | none       | boolean              | message reply icon                                                                                 |
| removeButton            | none       | boolean              | message remove icon                                                                                |
| status                  | none       | string               | message status info (waiting, sent, received, read)                                                |
| notch                   | true       | boolean              | message box notch                                                                                  |
| avatar                  | none       | url                  | message box avatar url                                                                             |
| renderAddCmp            | none       | function (component) | adding custom components to message box                                                            |
| copiableDate            | false      | boolean              | message box date text copiable                                                                     |
| focus                   | false      | boolean              | used in message focus feature in MessageList component, makes style of the component focused       |
| onMessageFocused        | none       | function             | makes focus value false after the message becomes focus                                            |
| reply                   | none       | object               | reply data                                                                                         |
| retracted               | none       | boolean              | message deleted or retracted                                                                       |

## Reply Message Component

```javascript
import { MessageBox } from 'react-chat-elements'
;<MessageBox
  reply={{
    photoURL: 'https://facebook.github.io/react/img/logo.svg',
    title: 'elit magna',
    titleColor: '#8717ae',
    message: 'Aliqua amet incididunt id nostrud',
  }}
  onReplyMessageClick={() => console.log('reply clicked!')}
  position={'left'}
  type={'text'}
  text={'Tempor duis do voluptate enim duis velit veniam aute ullamco dolore duis irure.'}
/>
```

![reply-message](https://user-images.githubusercontent.com/15075759/80224625-9dbbeb00-8652-11ea-994f-022be0cffe30.png)

## MeetingMessage Component

```javascript
import { MeetingMessage } from 'react-chat-elements'

<MeetingMessage
    subject={'New Release'}
    title={'in ullamco'}
    date={new Date()}
    collapseTitle={'Commodo aliquip'}
    participants={[
        {
            id: '1',
            title: 'Facebook',
        },
        .
        .
        .
    ]}
    dataSource={[
        {
            id: '1',
            avatar: 'https://facebook.github.io/react/img/logo.svg',
            message: 'Lorem ipsum dolor sit amet.',
            title: 'Elit magna',
            avatarFlexible: true,
            date: new Date(),
            event: {[
                title: 'Toplantı sona erdi!',
                avatars={[
                    src: 'https://facebook.github.io/react/img/logo.svg'
                ]}
            ]}
            record: {[
                avatar: 'https://facebook.github.io/react/img/logo.svg',
                title: 'Arama',
                savedBy: 'Kaydeden: Elit magna',
                time: new Date(),
            ]}
        },
        .
        .
        .
    ]} />
```

#### MeetingMessage props

| prop                    | default    | type     | description                                                                                        |
| ----------------------- | ---------- | -------- | -------------------------------------------------------------------------------------------------- |
| subject                 | none       | string   | Meeting messagee                                                                                   |

CHANGELOG.md
<details open>
<summary>12.0.4 Updates</summary>
<br>

# Meeting Link changes

## You can render custom meeting link action buttons

### New usage;

```js

import { MeetlingLink } from 'react-chat-elements'

<MeetlingLink
	actionButtons={[
		{
			onClickButton: test => {
				console.log(test)
			},
			Component: () => <div>Test</div>,
		},
		{
			onClickButton: test => {
				console.log(test)
			},
			Component: () => <div>Test</div>,
		},
	]}
	.
	.
	.
/>
```

### In MessageList component usage;

```js
import { MessageList } from 'react-chat-elements'

<MessageList
	actionButtons={[
		{
			onClickButton: test => {
				console.log(test)
			},
			Component: () => <div>Test</div>,
		},
		{
			onClickButton: test => {
				console.log(test)
			},
			Component: () => <div>Test</div>,
		},
	]}
	.
	.
	.
/>
```

</details>

<br>

<details >
<summary>12.0.0 Updates</summary>
<br>

# typescript support

</details>
<br>

<details>
<summary>11.0.0 Updates</summary>
<br>

This update target to fix component ref broken problems

Fixed issues:

- https://github.com/Detaysoft/react-chat-elements/issues/158
- https://github.com/Detaysoft/react-chat-elements/issues/157
- https://github.com/Detaysoft/react-chat-elements/issues/142

1.  All react-chat-elements components turneded to function component for "ref" property problems.

2.  In the [Input](https://github.com/Detaysoft/react-chat-elements#input-component) component `referance={...}` instead of use `ref={...}`

3.  **10.16.2** and before vesion usage

    **Before usage**: this.refs.input.clear();

    After **11.0.0** version usage is:

    ```js
    import { Input } from 'react-chat-elements'

    <Input
    	ref='input'
    	placeholder="Type here..."
    	multiline={true}
    	.
    	.
    	.
    />
    ```

    New usage in **class component**: clearRef();

    ```js

    import { Input } from 'react-chat-elements'
    let clearRef = () => {};
    this.inputReferance = React.createRef();

    <Input
    	referance={this.inputReferance}
    	clear={(clear) => clearRef = clear}
    	placeholder="Type here..."
    	multiline={true}
    	.
    	.
    	.
    />

    ```

    New usage in **function component**: clearRef();

    ```js

    import { Input } from 'react-chat-elements'
    let clearRef = () => {};
    const inputReferance = React.useRef();

    <Input
    	referance={inputReferance}
    	clear={(clear) => clearRef = clear}
    	placeholder="Type here..."
    	multiline={true}
    	.
    	.
    	.
    />

    ```

4.  In the [MessageList](https://github.com/Detaysoft/react-chat-elements#messagelist-component) component usage `referance={...}` instead of use `ref={...}`

    **Class Component:**

    ```js
    import { MessageList } from 'react-chat-elements'
    this.messageList = React.createRef();

    <MessageList
    	referance={this.messageList}
    	className='message-list'
    	lockable={true}
    	toBottomHeight={'100%'}
    	dataSource={[
    		{
    			position: 'right',
    			type: 'text',
    			text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
    			date: new Date(),
    		},
    		.
    		.
    		.
    	]} />
    ```

    **Function Component:**

    ```js
    import { MessageList } from 'react-chat-elements'
    const messageListReferance = React.useRef();

    <MessageList
    	referance={messageListReferance}
    	className='message-list'
    	lockable={true}
    	toBottomHeight={'100%'}
    	dataSource={[
    		{
    			position: 'right',
    			type: 'text',
    			text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
    			date: new Date(),
    		},
    		.
    		.
    		.
    	]} />
    ```

</details>


example/App.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
exports.__esModule = true;
var react_1 = require("react");
var src_1 = require("../src");
var fa_1 = require("react-icons/fa");
var md_1 = require("react-icons/md");
var lorem_ipsum_1 = require("lorem-ipsum");
var identicon_js_1 = require("identicon.js");
var clearRef = function () { };
function useForceUpdate() {
    var _a = (0, react_1.useState)(0), value = _a[0], setValue = _a[1];
    return function () { return setValue(function (value) { return value + 1; }); };
}
;
var App = function (props) {
    var messageListReferance = (0, react_1.useRef)();
    var inputReferance = (0, react_1.useRef)();
    var _a = (0, react_1.useState)(true), show = _a[0], setShow = _a[1];
    var _b = (0, react_1.useState)('chat'), list = _b[0], setList = _b[1];
    var _c = (0, react_1.useState)([]), messageList = _c[0], setMessageList = _c[1];
    var _d = (0, react_1.useState)([]), chatSource = _d[0], setChatSource = _d[1];
    var _e = (0, react_1.useState)([]), meetingSource = _e[0], setMeetingSource = _e[1];
    var _f = (0, react_1.useState)(false), isShowChild = _f[0], setIsShowChild = _f[1];
    var _g = (0, react_1.useState)(false), preview = _g[0], setPreview = _g[1];
    (0, react_1.useEffect)(function () {
        addMessage(8);
        var arr = [];
        for (var i = 0; i < 5; i++)
            arr.push(i);
        setChatSource(arr.map(function (x) { return random('chat', token()); }));
        setMeetingSource(arr.map(function (x) { return random('meeting', token()); }));
    }, []);
    var getRandomColor = function () {
        var letters = '0123456789ABCDEF';
        var color = '#';
        for (var i = 0; i < 6; i++) {
            color += letters[Math.floor(Math.random() * 16)];
        }
        return color;
    };
    var token = function () {
        return (Math.floor(Math.random() * 10 % 8));
    };
    var photo = function (size) {
        return new identicon_js_1["default"](String(Math.random()) + String(Math.random()), {
            margin: 0,
            size: size || 20
        }).toString();
    };
    var random = function (type, mtype) {
        switch (type) {
            case 'message':
                mtype = mtype || token();
                var status = 'waiting';
                switch (mtype) {
                    case 0:
                        mtype = 'photo';
                        status = 'sent';
                        break;
                    case 1:
                        mtype = 'file';
                        status = 'sent';
                        break;
                    case 2:
                        mtype = 'system';
                        status = 'received';
                        break;
                    case 3:
                        mtype = 'location';
                        break;
                    case 4:
                        mtype = 'spotify';
                        break;
                    case 5:
                        mtype = 'meeting';
                        break;
                    case 6:
                        mtype = 'video';
                        status = 'sent';
                        break;
                    case 7:
                        mtype = 'audio';
                        break;
                    case 8:
                        mtype = 'meetingLink';
                        break;
                    default:
                        mtype = 'text';
                        status = 'read';
                        break;
                }
                return {
                    position: (token() >= 1 ? 'right' : 'left'),
                    forwarded: true,
                    replyButton: true,
                    removeButton: true,
                    retracted: false,
                    reply: token() >= 1 ? ({
                        photoURL: token() >= 1 ? "data:image/png;base64,".concat(photo(150)) : null,
                        title: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                        titleColor: getRandomColor(),
                        message: (0, lorem_ipsum_1["default"])({ count: 1, units: 'sentences' })
                    }) : null,
                    meeting: token() >= 1 ? ({
                        subject: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                        title: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                        date: +new Date(),
                        collapseTitle: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                        participants: Array(token() + 6).fill(1).map(function (x) { return ({
                            id: Math.floor(Math.random() * 10 % 7),
                            title: (0, lorem_ipsum_1["default"])({ count: 1, units: 'words' })
                        }); }),
                        dataSource: Array(token() + 5).fill(1).map(function (x) { return ({
                            id: String(Math.random()),
                            avatar: "data:image/png;base64,".concat(photo(20)),
                            message: (0, lorem_ipsum_1["default"])({ count: 1, units: 'sentences' }),
                            title: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                            avatarFlexible: true,
                            date: +new Date(),
                            event: {
                                title: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                                avatars: Array(token() + 2).fill(1).map(function (x) { return ({
                                    src: "data:image/png;base64,".concat(photo(20)),
                                    title: "react, rce"
                                }); }),
                                avatarsLimit: 5
                            },
                            record: {
                                avatar: "data:image/png;base64,".concat(photo(20)),
                                title: (0, lorem_ipsum_1["default"])({ count: 1, units: 'words' }),
                                savedBy: 'Kaydeden: ' + (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                                time: new Date().toLocaleString()
                            }
                        }); })
                    }) : null,
                    type: mtype,
                    theme: 'white',
                    view: 'list',
                    title: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                    titleColor: getRandomColor(),
                    text: mtype === 'spotify' ? 'spotify:track:0QjjaCaXE45mvhCnV3C0TA' : (0, lorem_ipsum_1["default"])({ count: 1, units: 'sentences' }),
                    data: {
                        videoURL: token() >= 1 ? 'https://www.w3schools.com/html/mov_bbb.mp4' : 'http://www.exit109.com/~dnn/clips/RW20seconds_1.mp4',
                        audioURL: 'https://www.w3schools.com/html/horse.mp3',
                        uri: "data:image/png;base64,".concat(photo(150)),
                        status: {
                            click: true,
                            loading: 0.5,
                            download: mtype === 'video'
                        },
                        size: "100MB",
                        width: 300,
                        height: 300,
                        latitude: '37.773972',
                        longitude: '-122.431297',
                        staticURL: 'https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/pin-s-circle+FF0000(LONGITUDE,LATITUDE)/LONGITUDE,LATITUDE,ZOOM/270x200@2x?access_token=KEY'
                    },
                    onLoad: function () {
                        console.log('Photo loaded');
                    },
                    status: status,
                    date: +new Date(),
                    onReplyMessageClick: function () {
                        console.log('onReplyMessageClick');
                    },
                    onRemoveMessageClick: function () {
                        console.log('onRemoveMessageClick');
                    },
                    avatar: "data:image/png;base64,".concat(photo(20))
                };
            case 'chat':
                return {
                    id: String(Math.random()),
                    avatar: "data:image/png;base64,".concat(photo(20)),
                    avatarFlexible: true,
                    statusColor: 'lightgreen',
                    statusColorType: Math.floor(Math.random() * 100 % 2) === 1 ? 'encircle' : undefined,
                    alt: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                    title: (0, lorem_ipsum_1["default"])({ count: 2, units: 'words' }),
                    date: new Date(),
                    subtitle: (0, lorem_ipsum_1["default"])({ count: 1, units: 'sentences' }),
                    unread: Math.floor(Math.random() * 10 % 3),
                    muted: Math.floor(Math.random() * 10 % 2) === 1,
                    showMute: Math.floor(Math.random() * 10 % 2) === 1,
                    showVideoCall: Math.floor(Math.random() * 10 % 2) === 1,
                    dropdownMenu: (<src_1.Dropdown animationPosition="norteast" title='Dropdown Title' buttonProps={{
                            type: "transparent",
                            color: "#cecece",
                            icon: {
                                component: <md_1.MdOutlineMoreVert />,
                                size: 24
                            }
                        }} items={[
                            {
                                icon: {
                                    component: <fa_1.FaSquare />,
                                    float: 'left',
                                    color: 'red',
                                    size: 22
                                },
                                text: 'Menu Item'
                            },
                            {
                                icon: {
                                    component: <fa_1.FaSquare />,
                                    float: 'left',
                                    color: 'purple',
                                    size: 22
                                },
                                text: 'Menu Item'
                            },
                            {
                                icon: {
                                    component: <fa_1.FaSquare />,
                                    float: 'left',
                                    color: 'yellow',
                                    size: 22
                                },
                                text: 'Menu Item'
                            },
                        ]}/>)
                };
            case 'meeting':
                return {
                    id: String(Math.random()),
                    lazyLoadingImage: "data:image/png;base64,".concat(photo(20)),
                    avatarFlexible: true,
                    subject: (0, lorem_ipsum_1["default"])({ count: 1, units: 'sentences' }),
                    date: new Date(),
                    avatars: Array(token() + 2).fill(1).map(function (x) { return ({
                        src: "data:image/png;base64,".concat(photo(20))
                    }); }),
                    closable: true
                };
        }
    };
    var forceUpdate = useForceUpdate();
    var addMessage = function (mtype) {
        var list = messageList;
        list.push(random('message', mtype));
        setMessageList(list);
        clearRef();
        forceUpdate();
    };
    return (<div className='container'>
      <div className='chat-list'>
        <src_1.SideBar top={<div>
              <src_1.Popup 
        // show={show}
        header='Lorem ipsum dolor sit amet.' headerButtons={[{
                    type: 'transparent',
                    color: 'black',
                    onClick: function () {
                        setShow(false);
                    },
                    icon: {
                        component: <fa_1.FaTimes />,
                        size: 18
                    }
                }]} text='Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptatem animi veniam voluptas eius!' footerButtons={[{
                    color: 'white',
                    backgroundColor: '#ff5e3e',
                    text: "Vazgeç"
                }, {
                    color: 'white',
                    backgroundColor: 'lightgreen',
                    text: "Tamam"
                }]}/>

              <src_1.Button type='transparent' color='black' text={list === 'chat' ? 'MeetingList' : 'ChatList'} onClick={function () {
                setList(list === 'chat' ? 'meeeting' : 'chat');
            }}/>
            </div>} center={list === 'chat' ?
            <src_1.ChatList dataSource={chatSource} onClickMute={function (_a) {
                var props = __rest(_a, []);
                return console.log(props);
            }} onClickVideoCall={function (_a) {
                var props = __rest(_a, []);
                return console.log(props);
            }} onDragEnter={function (e, id) { return console.log(id, 'onDragEnter'); }} onDragLeave={function (e, id) { return console.log(id, 'onDragLeave'); }} onDrop={function (e, id) { return console.log(e, id, 'onDrop'); }} onDragComponent={function (id) { return <div className="on-drag-mlist">{(0, lorem_ipsum_1["default"])({ count: 4, units: 'words' })}</div>; }}/>
            :
                <src_1.MeetingList onMeetingClick={console.log} onShareClick={console.log} dataSource={meetingSource}/>} bottom={<span>
              <src_1.Button type='transparent' color='black' icon={{
                component: <fa_1.FaComments />,
                size: 18
            }}/>
              <src_1.Button type='transparent' color='black' icon={{
                component: <fa_1.FaSearch />,
                size: 18
            }}/>
              <src_1.Button text="Count"></src_1.Button>
            </span>}/>
    </div>
    <div className='right-panel'>
      <src_1.MessageList referance={messageListReferance} className='message-list' lockable={true} downButtonBadge={10} dataSource={messageList} sendMessagePreview={true} isShowChild={isShowChild} customProps={{
            onDragEnter: function (e) {
                e.preventDefault();
                console.log('onDragEnter');
                setIsShowChild(true);
            }
        }}>
        {preview ?
            <div className="on-drag-mlist" onClick={function () {
                    setIsShowChild(false);
                    setPreview(false);
                }}>preview click and finish</div>
            :
                <div className="on-drag-mlist" onDragOver={function (e) {
                        e.preventDefault();
                        console.log('onDragOver');
                    }} onDragLeave={function (e) {
                        e.preventDefault();
                        console.log('onDragLeave');
                        setIsShowChild(false);
                    }} onDrop={function (e) {
                        e.preventDefault();
                        console.log(e.dataTransfer.files, 'onDrop');
                        setPreview(true);
                    }}>
              {(0, lorem_ipsum_1["default"])({ count: 4, units: 'words' })}
          </div>}
      </src_1.MessageList>

      <src_1.Input placeholder="Mesajınızı buraya yazınız." defaultValue="" referance={inputReferance} clear={function (clear) { return clearRef = clear; }} 
    // buttonsFloat='left'
    onKeyPress={function (e) {
            if (e.shiftKey && e.charCode === 13) {
                return true;
            }
            if (e.charCode === 13) {
                clearRef();
                addMessage(8);
            }
        }} rightButtons={<src_1.Button text='Gönder' onClick={function () { return addMessage(8); }}/>}/>
      </div>
    </div>);
};
exports["default"] = App;
